mustache.go

```diff
@@ -159,7 +180,14 @@ type testNode struct {
 	elems     []node
 }
 
-func (n *testNode) render(t *Template, w *writer, c ...interface{}) error {
+var _ node = (*testNode)(nil)
+
+func (n *testNode) render(t *Template, w *writer, visitedNodes map[node]struct{}, c ...interface{}) error {
+	if err := tryAddNode(visitedNodes, n); err != nil {
+		return err
+	}
+	defer delete(visitedNodes, n)
```

Comment:
this feels a little weird to me. we're using the caller's map as a mutable scratchpad?

Reply:
yes, since the library is doing a DFS traversal of the mustache node graph using the render(...) method, I wasn't sure of a better way to detect the cycle other than this way.